### PR TITLE
Improve attribute loading

### PR DESCRIPTION
**Pull request by VivekVinushanth**
### Related Issues
- https://github.com/wso2/product-is/issues/15485 (Wait till BE support is there)

### CODE REVIEW

**Review by brionmario — DISMISSED** [review]

File: modules/core/src/models/claim.ts [hunk 1]
```diff
@@ -31,9 +31,20 @@ export interface Claim {
     required: boolean;
     supportedByDefault: boolean;
     attributeMapping: AttributeMapping[];
+    associatedExternalClaims?: AssociatedExternalClaim[];
+
```
Comment (thivi) [inline modules/core/src/models/claim.ts:35]:
```suggestion
```

**Review by thivi — APPROVED** [review]

File: apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx [hunk 2]
```diff
@@ -446,7 +438,7 @@ export const EditBasicDetailsLocalClaims: FunctionComponent<EditBasicDetailsLoca
                                 label={ t("console:manage.features.claims.local.forms.supportedByDefault.label") }
                                 required={ false }
                                 defaultValue={ claim?.supportedByDefault }
-                                listen={ (values) => {
+                                listen={ (values: any) => {
```
Comment (thivi) [inline apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx:441]: Let's avoid using the `any` type.

File: apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx [hunk 3]
```diff
@@ -499,7 +491,7 @@ export const EditBasicDetailsLocalClaims: FunctionComponent<EditBasicDetailsLoca
                                 data-testid={ `${ testId }-form-required-checkbox` }
                                 readOnly={ isReadOnly }
                                 hint={ t("console:manage.features.claims.local.forms.requiredHint") }
-                                listen ={ (value) => {
+                                listen ={ (value: any) => {
```
Comment (thivi) [inline apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx:494]: Let's avoid the `any` type.

File: apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx [hunk 4]
```diff
@@ -529,7 +521,7 @@ export const EditBasicDetailsLocalClaims: FunctionComponent<EditBasicDetailsLoca
                                 data-testid={ `${ testId }-form-readonly-checkbox` }
                                 readOnly={ isReadOnly }
                                 hint={ t("console:manage.features.claims.local.forms.readOnlyHint") }
-                                listen={ (value) => {
+                                listen={ (value: any) => {
```
Comment (thivi) [inline apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx:524]: Let's avoid the `any` type.

File: apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx [hunk 5]
```diff
@@ -259,7 +251,7 @@ export const EditBasicDetailsLocalClaims: FunctionComponent<EditBasicDetailsLoca
             });
     };
 
-    const onSubmit = (values) => {
+    const onSubmit = (values: any) => {
```
Comment (thivi) [inline apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx:254]: Let's use Record<string, any>

**Review by brionmario — APPROVED** [review]

File: apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx [hunk 6]
```diff
@@ -91,10 +92,10 @@ export const EditBasicDetailsLocalClaims: FunctionComponent<EditBasicDetailsLoca
     const [ hasMapping, setHasMapping ] = useState<boolean>(false);
     const [ mappingChecked, setMappingChecked ] = useState<boolean>(false);
 
-    const nameField = useRef<HTMLElement>(null);
-    const regExField = useRef<HTMLElement>(null);
-    const displayOrderField = useRef<HTMLElement>(null);
-    const descriptionField = useRef<HTMLElement>(null);
+    const nameField: React.MutableRefObject<HTMLElement> = useRef<HTMLElement>(null);
```
Comment (brionmario) [inline apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx:95]: Let's explicitly import `MutableRefObject` from "react".
```suggestion
    const nameField: MutableRefObject<HTMLElement> = useRef<HTMLElement>(null);
```

File: apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx [hunk 7]
```diff
@@ -207,7 +199,7 @@ export const EditBasicDetailsLocalClaims: FunctionComponent<EditBasicDetailsLoca
                     message: t("console:manage.features.claims.local.notifications.deleteClaim.success.message")
                 }
             ));
-        }).catch(error => {
+        }).catch((error: any) => {
```
Comment (brionmario) [inline apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx:202]:
```suggestion
        }).catch((error: AxiosError) => {
```

**Comment by VivekVinushanth** [comment]
Will send a separate PR fixing these lint issues.

**Pull request merged**